### ISSUE TITLE
[Bug]: Observer options not taken into account

### ISSUE THREAD

**Issue by 1u0n** [DaveKeehl/svelte-reveal#149]
### Describe the bug

when creating the observer like 

```
<div use:reveal={{ transition: "fly", debug: true, ref: "a", threshold: 0.2, marginBottom: 100, marginTop: 100 }}>
```
the lib options (transition, x, debug, ref, ...) are correctly used, but the Observer options (threshold, marginBottom, ...) are ignored.

This is I guess due to passing `config.observer` directly in `createObserver()` instead of creating an object with the user's parameters.

Changing the Observer options through the API (setObserverThreshold, setObserverRootMargin, ...) works fine.

### Reproduction

```
<div use:reveal={{ transition: "fly", debug: true, ref: "a", threshold: 0.2, marginBottom: 100, marginTop: 100 }}>
```

the applied threshold will be the default 0.6 instead of the desired 0.2. Same for margins and root.

### System Info

```shell
this is not system-dependant
```

### Severity

Serious, but I can work around it

**Comment by DaveKeehl** [comment]
Thanks for reporting it. Will look into it!

**Comment by DaveKeehl** [comment]
Yes, you were right. It was because the `createObserver` function was using the default options instead of the ones overridden by the user. I've managed to fix this, but I have also identified other small areas of improvements related to this, so I still want to take a few days to work on it.

Expect a new minor release in the upcoming days.

**Comment by DaveKeehl** [comment]
I just released version `0.7.0`. Feel free to re-open this issue or create a new one in case of problems!
